hardware_interface/src/resource_manager.cpp

```diff
@@ -1770,7 +1770,6 @@ void ResourceManager::shutdown_async_components()
 HardwareReadWriteStatus ResourceManager::read(
   const rclcpp::Time & time, const rclcpp::Duration & period)
 {
-  std::lock_guard<std::recursive_mutex> guard(resources_lock_);
```

Comment:
why don't we need this here? or why do we need this lock in other methods of this file?

Reply:
The mutex from here is from the Resource Manager, it is used along with non-RT threads, as it is used this way, there is most likely a chance that it can block the read and write cycles.

We have added in other methods because, when this happens, the read cycles of that particular interface can be skipped instead of waiting for the cycle as it is configuring itself to be activated or deactivated